### DISCUSSION TITLE
Strange paths for custom shaders

### DISCUSSION THREAD

**Discussion by arithefirst** [ghostty-org/ghostty#3304] [Feature Requests, Ideas] [answered]
There's already a [discussion in the Discord](https://discord.com/channels/1005603569187160125/1322071404568645723) about this, but it has to do with ghostty's weird paths for custom shaders. For some reason, anytime I added a custom shader to my config, it would prepend `~/.config/ghostty/` to itself

See below images
![image](https://github.com/user-attachments/assets/01beb901-6a6d-4bee-abf8-29a5d9a99043)
![image](https://github.com/user-attachments/assets/ca9d625f-35b9-487d-bccb-c1c89f75396a)

Not sure if this is an intended feature, but I figured I would create an issue in case this _is_ a bug,

**Comment by mitchellh** [comment]
Not a bug! We don’t expand tilde because that’s a shell thing. But we could…. Let me consider this and I’ll turn it into an issue

**Reply by arithefirst** [reply]
The issue was less not expanding the tilde, it's that ghostty prepends `/home/$USER/.config/ghostty/` to the beginning of any path. Even when I use absolute paths instead of the tilde the same behavior occurs

**Reply by mitchellh** [reply]
That shouldn’t be the case. It should only do this for relative paths. I’ll check.

in this case though, the tilde makes this relative. But I’ll try others when I get to a computer

**Reply by arithefirst** [reply]
Another issue is that it seems to be very inconsistent, it worked as intended for about an hour, and then for about an hour it had this behavior, then went back. It's very odd, and may end up just being an issue with my computer for some reason

**Comment by walbarellos** [comment]
ChatGPT was suggesting shader-theme /location/ (after seeing your image I tried: custom-shader, it worked!!! (using tumbleweed.)

**Comment by mitchellh** [comment, marked as answer]
I haven't heard any other reports of this, and custom shaders have had significant changes since this was reported. If you're still having issues please open a new discussion.
